src/eigenlayer/ObolEigenLayerPodController.sol

```diff
@@ -4,173 +4,155 @@ pragma solidity 0.8.19;
-
-        feeRecipient = recipient;
-        feeShare = share;
-        eigenLayerDelegationManager = delegationManager;
-        eigenLayerPodManager = IEigenPodManager(eigenPodManager);
-        delayedWithdrawalRouter = IDelayedWithdrawalRouter(withdrawalRouter);
+  /// @dev returned on failed call
+  error CallFailed(bytes data);
+  /// @dev If Invalid fee setup
+  error Invalid_FeeSetup();
+  /// @dev Invalid fee share
+  error Invalid_FeeShare();
+  /// @dev user unauthorized
+  error Unauthorized();
+  /// @dev contract already initialized
+  error AlreadyInitialized();
+
+  /// @dev Emiited on intialize
+  event Initialized(address eigenPod, address owner);
+
+  /// -----------------------------------------------------------------------
+  /// libraries
+  /// -----------------------------------------------------------------------
+  using SafeTransferLib for address;
+  using SafeTransferLib for ERC20;
+
+  uint256 internal constant PERCENTAGE_SCALE = 1e5;
+
+  /// -----------------------------------------------------------------------
+  /// storage - immutables
+  /// -----------------------------------------------------------------------
+
+  /// @notice address of Eigenlayer delegation manager
+  /// @dev This is the address of the delegation manager transparent proxy
+  address public immutable eigenLayerDelegationManager;
+
+  /// @notice address of EigenLayerPod Manager
+  /// @dev this is the pod manager transparent proxy
+  IEigenPodManager public immutable eigenLayerPodManager;
+
+  /// @notice address of delay withdrawal router
+  IDelayedWithdrawalRouter public immutable delayedWithdrawalRouter;
+
+  /// @notice fee address
+  address public immutable feeRecipient;
+
+  /// @notice fee share. Represented as an integer from 1->10000 (100%)
+  uint256 public immutable feeShare;
+
+  /// -----------------------------------------------------------------------
+  /// storage
+  /// -----------------------------------------------------------------------
+
+  /// @notice address of deployed Eigen pod
+  address public eigenPod;
+
+  /// @notice address of a withdrawalAddress
+  address public withdrawalAddress;
+
+  /// @notice address of owner
+  address public owner;
+
+  modifier onlyOwner() {
+    if (msg.sender != owner) revert Unauthorized();
+    _;
+  }
+
+  constructor(
+    address recipient,
+    uint256 share,
+    address delegationManager,
+    address eigenPodManager,
+    address withdrawalRouter
+  ) {
+    if (recipient != address(0) && share == 0) revert Invalid_FeeSetup();
+    if (share > PERCENTAGE_SCALE) revert Invalid_FeeShare();
+
+    feeRecipient = recipient;
+    feeShare = share;
+    eigenLayerDelegationManager = delegationManager;
+    eigenLayerPodManager = IEigenPodManager(eigenPodManager);
+    delayedWithdrawalRouter = IDelayedWithdrawalRouter(withdrawalRouter);
+  }
+
+  /// @dev Enables contract to receive ETH
+  // defined on the clone implementation
+  // receive() external payable {}
```

Comment:
can be removed

Reply:
this was added for documentation purposes. if another dev takes a look they can understand why receive isn't defined here